### PR TITLE
Add documentation for rubygems.org rate limits

### PR DESCRIPTION
**Pull request by sonalkr132**
My primary goal with this is to having a link (besides code) to share when someone complains about rate limit, specifically `gem push`.
Perhaps this doesn't have to be this verbose, please feel free to share your opinion on this.

### CODE REVIEW

File: rubygems-org-rate-limits.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+* User sign up - `POST /users`
+* Password reset request - `POST /passwords`
+* Profile update - `PATCH /profile`
+* Profile delete - `DELETE /profile`
+* Email confirmation request - `POST /email_confirmations`
+
+## Rate limits with exponential backoff 
+
+It is not possible to brute force your MFA code in a single time window. However, an attacker's chance of successfully guessing the code at least once increases when the brute force is attempted over an extended period. You can read more about this [here](https://security.stackexchange.com/a/185917) and check our calculation of the backoff period [here](https://github.com/rubygems/rubygems.org/pull/2330#issuecomment-643931531).
+Following endpoints have rate limits of **300 requests/5 miinutes** and **600 requests/25 hours**:
```
Comment (deivid-rodriguez) [inline rubygems-org-rate-limits.md:36]:
```suggestion
Following endpoints have rate limits of **300 requests/5 minutes** and **600 requests/25 hours**:
```

File: rubygems-org-rate-limits.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+---
+layout: default
+title: RubyGems.org rate limits
+url: /rubygems-org-rate-limits
+previous: /rubygems-org-api-v2
+next: /run-your-own-gem-server
+---
+
+<em class="t-gray">Why are you see seeing 429 responses?</em>
```
Comment (qrush) [inline rubygems-org-rate-limits.md:9]: "Why are you seeing"

File: rubygems-org-rate-limits.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+Loadbalancer rate limits
```
Comment (qrush) [inline rubygems-org-rate-limits.md:11]: "Load balancer" is usually 2 separate words.

File: rubygems-org-rate-limits.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+## Miscellaneous rate limits
+
+* 10 request/10 minutes on gem yank - `DELETE /api/v1/gems/yank`
+* 100 request/10 minutes/email or username on sign in - `POST /session`
+* 100 request/10 minutes/email or username on API key show - `GET /api/v1/api_key`
+* 10 request/10 minutes/email on password reset request - `POST /passwords`
+* 10 request/10 minutes/email on email confirmation request - `POST /email_confirmations`
+
+The RubyGems.org team may occasionally blackhole user IP addresses for extreme cases to protect the platform. If you think this has happened to you, please [submit a help ticket](http://help.rubygems.org/discussion/new), and we'll be happy to look at it.
```
Comment (qrush) [inline rubygems-org-rate-limits.md:62]: https:// here?

**Comment by sonalkr132** [comment]
Thank you for your review. I have fixed the typos. 

> I wonder if we should return a URL to this guide in the API response when you're getting rate limited

Not a bad idea. Do you know any API which already does this?

**Pull request merged**